.github/workflows/ci.yml

```diff
@@ -33,26 +33,27 @@ jobs:
           elixir-version: ${{matrix.pair.elixir}}
 
       - uses: actions/cache@v2
+        id: cache-deps
         with:
           path: |
             deps
             _build
           key: mix-${{ runner.os }}-${{matrix.pair.elixir}}-${{matrix.pair.otp}}-${{ hashFiles('**/mix.lock') }}
 
-      - run: mix deps.get
+      - run: mix do deps.get, deps.compile
+        if: steps.cache-deps.outputs.cache-hit != 'true'
 
       - run: mix format --check-formatted
         if: ${{ matrix.lint }}
 
       - run: mix deps.unlock --check-unused
         if: ${{ matrix.lint }}
 
-      - run: mix deps.compile
-
       - run: mix compile --warnings-as-errors
         if: ${{ matrix.lint }}
 
       - run: mix test
+        if: ${{!matrix.coverage}}
```

Comment:
`coverall.github` also runs tests, so no point in running tests if we're running test coverage. 🙃

Reply:
For me we can remove coverall.

Reply:
@josevalim I personally love it 😄, but sure, next step if you want!

Reply:
Ah, so let's stick with it. your call!